enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -369,8 +365,9 @@ def resolve_policy(cls, redeemable_policies):
         """
         # For now, we inefficiently make one call per subsidy record.
         subsidy_uuids = set(redeemable_policy.subsidy_uuid for redeemable_policy in redeemable_policies)
+        subsidy_client = get_versioned_subsidy_client()
         subsidy_balances = {
-            subsidy_uuid: cls.subsidy_client.retrieve_subsidy(subsidy_uuid)["current_balance"]
+            subsidy_uuid: subsidy_client.retrieve_subsidy(subsidy_uuid)["current_balance"]
```

Comment:
Without making this change to get the versions subsidy client (i.e., `cls.subsidy_client`), I'm seeing getting a "No attribute `retrieve_subsidy` exists on `subsidy_client`" type of error.

I can replicate this in the Django shell, where using the class method approach to access the `subsidy_client` property errors but using the `subsidy_client` method on a single instance of `SubsidyAccessPolicy` does have the `retrieve_subsidy` attribute.

Is there an alternative way of fixing this?

Reply:
You can decorate the property definition with a `@staticmethod` (and remove the `self` parameter from it).